assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -444,18 +447,27 @@ const baseSelectors = {
 		const { sharingSettings, savedSharingSettings } = state;
 
 		if ( keys ) {
-			return ! isEqual(
-				pick( sharingSettings, keys ),
-				pick( savedSharingSettings, keys )
-			);
+			let originalKeys = at( sharingSettings, keys );
+			if ( Array.isArray( originalKeys ) && originalKeys.length === 1 ) {
+				originalKeys = originalKeys[ 0 ];
+			}
+
+			let savedKeys = at( savedSharingSettings, keys );
+			if ( Array.isArray( savedKeys ) && savedKeys.length === 1 ) {
+				savedKeys = savedKeys[ 0 ];
+			}
+
+			if ( Array.isArray( originalKeys ) && Array.isArray( savedKeys ) ) {
+				originalKeys.sort();
+				savedKeys.sort();
+			}
+
+			return ! isEqual( originalKeys, savedKeys );
```

Comment:
What's the reason behind extracting the values in a one-item-array? It seems to add complexity to the code when something like: `isEqual( [ 'a' ], [ 'a' ] )` should work the same as `isEqual( 'a', 'a' )`, which is all this is doing.

What about simplifying this to:

```suggestion
			const originalKeys = at( sharingSettings, keys );
			const savedKeys = at( savedSharingSettings, keys );
			
			originalKeys.sort();
			savedKeys.sort();

			return ! isEqual( originalKeys, savedKeys );
```

Reply:
I've reverted this change, as `haveSharingSettingsChanged` is no longer being called with the `keys` argument, and it's not clear that we'd need this modification. The existing code which uses Lodash `pick`/`isEqual` seems in keeping with the non-keys version, and we can revisit this in future, if necessary, when we do start using `haveSharingSettingsChanged` with `keys`.